src/contracts/middleware/BLSSignatureChecker.sol

```diff
@@ -1,508 +1,218 @@
+                    nonSignerPubkeyHashes[i] = nonSignerStakesAndSignature.nonSignerPubkeys[i].hashG1Point();
+                    
+                    // check that the nonSignerPubkeys are sorted and free of duplicates
+                    if (i != 0) {
+                        require(uint256(nonSignerPubkeyHashes[i]) > uint256(nonSignerPubkeyHashes[i - 1]), "BLSSignatureChecker.checkSignatures: nonSignerPubkeys not sorted");
+                    }
+
+                    nonSignerQuorumBitmaps[i] = 
+                        registryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex(
+                            nonSignerPubkeyHashes[i], 
+                            referenceBlockNumber, 
+                            nonSignerStakesAndSignature.nonSignerQuorumBitmapIndices[i]
+                        );
+                    
+                    // subtract the nonSignerPubkey from the running apk to get the apk of all signers
+                    apk = apk.plus(
+                        nonSignerStakesAndSignature.nonSignerPubkeys[i]
+                            .negate()
+                            .scalar_mul_tiny(
+                                BitmapUtils.countNumOnes(nonSignerQuorumBitmaps[i] & signingQuorumBitmap) // we subtract the nonSignerPubkey from each quorum that they are a part of, TODO: 
+                            )
+                    );
                 }
             }
-            require(success, "BLSSignatureChecker.checkSignatures: non signer addition failed");
+            // loop through each quorum number
+            for (uint8 quorumNumberIndex = 0; quorumNumberIndex < quorumNumbers.length;) {
+                // get the quorum number
+                uint8 quorumNumber = uint8(quorumNumbers[quorumNumberIndex]);
+                // get the totalStake for the quorum at the referenceBlockNumber
+                quorumStakeTotals.totalStakeForQuorum[quorumNumberIndex] = 
+                    stakeRegistry.getTotalStakeAtBlockNumberFromIndex(quorumNumber, referenceBlockNumber, nonSignerStakesAndSignature.totalStakeIndices[quorumNumberIndex]);
+                // copy total stake to signed stake
+                quorumStakeTotals.signedStakeForQuorum[quorumNumberIndex] = quorumStakeTotals.totalStakeForQuorum[quorumNumberIndex];
+                // loop through all nonSigners, checking that they are a part of the quorum via their quorumBitmap
+                // if so, load their stake at referenceBlockNumber and subtract it from running stake signed
+                for (uint32 i = 0; i < nonSignerStakesAndSignature.nonSignerPubkeys.length; i++) {
+                    // keep track of the nonSigners index in the quorum
+                    uint32 nonSignerForQuorumIndex = 0;
+                    // if the nonSigner is a part of the quorum, subtract their stake from the running total
+                    if (nonSignerQuorumBitmaps[i] >> quorumNumber & 1 == 1) {
```

Comment:
I see this kind of operation a lot, including -- I think -- with the operations "reversed", i.e. something like
```
if (nonSignerQuorumBitmaps[i] && (1 << quorumNumber) != 0) {
```
Feels like it would be worth it to define a utility and add it to the BitmapUtils library for this. am imagining something like
`function numberIsInBitmap(uint256 bitmap, uint8 numberToCheckForInclusion) returns (bool)`

Reply:
opened a small PR for this https://github.com/Layr-Labs/eigenlayer-contracts/pull/213

Reply:
PR was merged, resolving this now.